src/test/txvalidationcache_tests.cpp

```diff
@@ -105,9 +105,11 @@ BOOST_FIXTURE_TEST_CASE(tx_mempool_block_doublespend, TestChain100Setup)
 static void ValidateCheckInputsForAllFlags(const CTransaction &tx, uint32_t failing_flags, bool add_to_cache)
 {
     PrecomputedTransactionData txdata(tx);
-    // If we add many more flags, this loop can get too expensive, but we can
-    // rewrite in the future to randomly pick a set of flags to evaluate.
-    for (uint32_t test_flags=0; test_flags < (1U << 16); test_flags += 1) {
+
+    for (uint32_t count=0; count < 10000; count++) {
+        // Randomly selects flag combinations
+        uint32_t test_flags = rand() % (HIGHEST_FLAG << 1);
```

Comment:
we don't use `rand()` anywhere in the code, would be preferable to use a `FastRandomContext insecure_rand(true);` here, also for determinism - for reproducible it is undesirable for different test runs to test a different subset of flags

Reply:
Fixed. Thanks for the explanation!